visual_behavior/decoding_population/general_funs.py

```diff
@@ -0,0 +1,259 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Tue Oct 13 21:23:43 2019
+
+@author: farzaneh
+"""
+'''
+from importlib import reload
+import x
+reload(x)
+'''
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+def set_frame_window_flash_omit(time_win, samps_bef, frame_dur):
+    # Convert a time window (relative to trial onset) to frames units, relative to "trace" begining; i.e. index on the trace whose time 0 is trace[samps_bef]. 
+    # samps_bef: number of frames before image/omission
+    
+    import numpy as np
```

Comment:
Generally bad to include import statements within a function. Especially bad since numpy here is also imported in the entire module

Reply:
why should they not be included within a function? do you mean they should be written outside the function, before it starts?

Reply:
Lots of reasons to import outside of functions. And yes, I mean all the import statements should be at the top of the module. 

- If you have `import numpy as np` inside every function, then it gets imported over and over again, and is simply slow
- Its harder to update or debug the code if every function is importing different modules, which makes the code less transparent. 
- The python documentation says its best practice to have all import statements at the top of the module: https://docs.python.org/3/tutorial/modules.html